src/NRules/NRules.Json/Converters/ExpressionConverter.cs

```diff
@@ -472,4 +480,22 @@ private void WriteDefaultExpression(Utf8JsonWriter writer, JsonSerializerOptions
     {
         writer.WriteProperty(nameof(value.Type), value.Type, options);
     }
+
+    private BlockExpression ReadBlockExpression(ref Utf8JsonReader reader, JsonSerializerOptions options)
+    {
+        var type = reader.ReadProperty<Type>(nameof(BlockExpression.Type), options);
+        if (type is null)
```

Comment:
@gritcsenko I would like to not handle this null type as a one-off. There are many places where we read type and other required elements. I would remove this check, and then separately, can add ReadRequiredProperty method, or bake this check into the existing ReadProperty method

Reply:
I'll handle post-merge